src/main/java/ratpack/zipkin/ServerTracingModule.java

```diff
@@ -50,11 +49,28 @@ protected void configure() {
         .to(DefaultServerTracingHandler.class)
         .in(Singleton.class);
 
-    bind(HttpClient.class).annotatedWith(Zipkin.class)
-        .to(ZipkinHttpClientImpl.class)
-        .in(Singleton.class);
+    bind(ClientTracingInterceptor.class)
+            .to(DefaultClientTracingInterceptor.class)
+            .in(Singleton.class);
+
+    Provider<ClientTracingInterceptor> clientTracingInterceptorProvider =
+            getProvider(ClientTracingInterceptor.class);
+
+    Provider<HttpClient> httpClientProvider = () ->
+            // HttpClient.of throws exceptions but Providers can not, hence the unchecked wrapper.
+            Exceptions.uncheck(() -> HttpClient.of(spec -> {
```

Comment:
This isn't perfect, I think it needs to be additive right now meaning it can't take over the HttpClient directly yet without ratpack changes...